bin/runtime/src/lib.rs

```diff
@@ -640,6 +646,49 @@ impl pallet_utility::Config for Runtime {
     type PalletsOrigin = OriginCaller;
 }
 
+pub struct AlephChainExtension;
+impl ChainExtension<Runtime> for AlephChainExtension {
+    fn call<E: Ext>(func_id: u32, env: Environment<E, InitState>) -> Result<RetVal, DispatchError>
+    where
+        <E::T as SysConfig>::AccountId: UncheckedFrom<<E::T as SysConfig>::Hash> + AsRef<[u8]>,
+    {
+        match func_id {
+            41 => {
```

Comment:
Consider extracting to a const for better readability:
```rust
const SNARCOS_CHAIN_EXT: usize = 41;

match func_id {
  SNARCOS_CHAIN_EXT => { ... }
```

Reply:
surely, done

Reply:
I edited original post and suggested using an enum -- bit more futureproof if you add more extensions.

Reply:
Ah yes, I was looking at my older code but it had a `TryFrom` impl that used an external crate for `u32 -> enum` transformation.